content/4-component-composition/5-context/vue3/App.vue

```diff
@@ -0,0 +1,21 @@
+<script setup>
+import { ref, provide } from "vue";
+import UserProfile from "./UserProfile.vue";
+
+const user = ref({
+  id: 1,
+  username: "unicorn42",
+  email: "unicorn42@example.com",
+});
+
+function updateUsername(username) {
+  user.value.username = username;
+}
+
+provide("userContextProvider", { user, updateUsername });
```

Comment:
To match Svelte example, use context name `user` rather than `userContextProvider`.